### ISSUE TITLE
Errors on Import

### ISSUE THREAD

**Issue by lowlydba** [dataplat/dbaclone#191]
**Before submitting the bug**
- [x] Ensure you are able to reproduce it on the latest released version (_we release often_)
- [x] Verified this bug is not already reported in an issue
- [x] Verified errors are not related to permissions
- [x] Can reproduce in a clean PowerShell session (_clean = `powershell -NoProfile`_)

**Describe the bug**
I haven't been able to dig in much deeper yet, but after having used `Set-DcnConfiguration` to point to a valid dbaclone database, the following error appears on `Import-Module dbaclone`. It appears to work fine after the error and doesn't happen 100% of the time.

**To Reproduce**
```
PS C:\> Import-Module dbaclone -VErbose
VERBOSE: Loading module from path 'C:\Program Files\WindowsPowerShell\Modules\dbaclone\0.10.63\dbaclone.psd1'.
VERBOSE: Populating RepositorySourceLocation property for module dbaclone.
VERBOSE: Loading module from path 'C:\Program Files\WindowsPowerShell\Modules\dbaclone\0.10.63\dbaclone.psm1'.
WARNING: [13:57:06][Get-DcnClone] Something is wrong in the module configuration | Cannot convert value "xxxx.xx.com" to type "Sqlcollaborative.Dbatools.Parameter.DbaInstanceParameter". Error: "Failed to
interpret input as Instance: xxxx.xx.com"
Exception calling "InvokeScript" with "4" argument(s): "Exception calling "InvokeScript" with "4" argument(s): "Cannot convert value "xxxx.xx.com" to type
"Sqlcollaborative.Dbatools.Parameter.DbaInstanceParameter". Error: "Failed to interpret input as Instance: xxxx.xx.com"""
At C:\Program Files\WindowsPowerShell\Modules\dbaclone\0.10.63\dbaclone.psm1:49 char:9

```

**Expected behavior**
A clean import.

**Comment by blitzmann** [comment]
I can confirm that this has been an issue

**Comment by sanderstad** [comment]
With the latest version of dbaclone 0.10.65 (released today) this should be a thing of the past

**Comment by lowlydba** [comment]
Haven't tested extensively yet, but so far so good on the new version. Thanks!

**Comment by lowlydba** [comment]
Unfortunately, I'm still seeing this on a brand new server with v 0.10.68.

If `Set-DcnConfiguration` **has not** been run, the error received is below. If it **has** been run, the error is the same as the original posted.

```
WARNING: [10:30:03][Get-DcnClone] The module setup has NOT yet successfully run. Please run 'Set-DcnConfiguration'
WARNING: [10:30:03][Get-DcnImage] The module setup has NOT yet successfully run. Please run 'Set-DcnConfiguration'
Exception calling "InvokeScript" with "4" argument(s): "You cannot call a method on a null-valued expression."
At C:\Program Files\WindowsPowerShell\Modules\dbaclone\0.10.68\dbaclone.psm1:49 char:9
+     else { $ExecutionContext.InvokeCommand.InvokeScript($false, ([scr ...
+            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : RuntimeException

```

**Comment by ay-azara** [comment]
Alright, the issue still present but I do have a workaround for you.

## TLDR

Run `Set-PSFConfig -Module dbaclone -Name informationstore.mode -Value 'SQL'`.

## The long version

To recap, these are the errors and, for me, they started while I was trying to call `New-DcnImage`

```
WARNING: [15:19:54][Get-DcnClone] The module setup has NOT yet successfully run. Please run 'Set-DcnConfiguration'
WARNING: [15:19:54][Get-DcnImage] The module setup has NOT yet successfully run. Please run 'Set-DcnConfiguration'

The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Exception calling "InvokeScript" with "4"       
argument(s): "You cannot call a method on a null-valued expression."
Exception calling "InvokeScript" with "4" argument(s): "You cannot call a method on a null-valued expression."
```

This led to another issue where the module couldn't be imported.

```powershell
Set-DcnConfiguration : The 'Set-DcnConfiguration' command was found in the module 'dbaclone', but the module could not be loaded. For more information, run 
'Import-Module dbaclone'.
```

Looking at `$Error` it seemed to be for the same reason as above, just buried. The solution of running `Set-DcnConfiguration` became an impossible chicken/egg scenario.

So, time to open up VSCode, create a file with `Import-Module Dbaclone`, set the debugger on that line and start debugging.

We step inside `Import-Module` and keep stepping over until we hit an error. VSCode will automatically open up the Dbaclone files as we continue.

```powershell
dbaclone\0.10.72\dbaclone.psm1
L49: else { $ExecutionContext.InvokeCommand.InvokeScript($false, ([scriptblock]::Create([io.file]::ReadAllText($resolvedPath))), $null, $null) }
```

This statement is failing so we update the breakpoint and start over but this time we step inside and keep stepping over until we hit the next error.

```powershell
dbaclone\0.10.72\internal\scripts\postimport.ps1
L25: if (((Get-PSFConfigValue -FullName dbaclone.informationstore.mode).ToLower()) -eq 'file') {
```

As you can see, this function is leveraging the Powershell Framework's PSFConfig functions to get/set Dbaclone config settings. If we run the snippet `Get-PSFConfigValue -FullName dbaclone.informationstore.mode` we find that the value is empty. I don't know how this occurred and I'm unlikely to look deeper into it but the solution is clear, manually setting it with `Set-PSFConfig -Module dbaclone -Name informationstore.mode -Value 'SQL'`.

And there you have. In my case I'm using SQL. If you're using something else use that. I'm not sure how the module loads after install before you run `Set-DcnConfiguration` but that's a question for another day).

**Comment by lowlydba** [comment]
Thanks for the followup on this! I am no longer working where this was a problem, but this makes sense enough to unblock them I'm sure.

**Comment by NotPhoton** [comment]
Just for future refrence to anybody who is still having this issue, the SQL Instance in your Set-DCNConfiguration must be typed in all lowercase. New-DCNImage fails with the "Failed to interpret input as Instance" error otherwise. Set-DCNConfiguration will always work regardless of if the instance is all lowercase or not.
